hist/hist/src/TGraph.cxx

```diff
@@ -2456,35 +2457,35 @@ Double_t **TGraph::ShrinkAndCopy(Int_t size, Int_t oend)
 /// ~~~
 
 void TGraph::Sort(Bool_t (*greaterfunc)(const TGraph*, Int_t, Int_t) /*=TGraph::CompareX()*/,
-                  Bool_t ascending /*=kTRUE*/, Int_t low /* =0 */, Int_t high /* =-1111 */)
+                  Bool_t ascending /*=kTRUE*/, Int_t low /*=0*/, Int_t high /*=-1111*/)
 {
-
-   // set the bit in case of an ascending =sort in X
-   if (greaterfunc == TGraph::CompareX && ascending  && low == 0 && high == -1111)
+   // set the bit in case of an ascending =sort in X   
+   if (greaterfunc == TGraph::CompareX && ascending && low == 0 && high == -1111)
       SetBit(TGraph::kIsSortedX);
 
-   if (high == -1111) high = GetN() - 1;
-   //  Termination condition
-   if (high <= low) return;
-
-   int left, right;
-   left = low; // low is the pivot element
-   right = high;
-   while (left < right) {
-      // move left while item < pivot
-      while (left <= high && greaterfunc(this, left, low) != ascending)
-         left++;
-      // move right while item > pivot
-      while (right > low && greaterfunc(this, right, low) == ascending)
-         right--;
-      if (left < right && left < high && right > low)
-         SwapPoints(left, right);
-   }
-   // right is final position for the pivot
-   if (right > low)
-      SwapPoints(low, right);
-   Sort(greaterfunc, ascending, low, right - 1);
-   Sort(greaterfunc, ascending, right + 1, high);
+   if (high == -1111) high = fNpoints;
+   
+   // Create a vector to hold the indices of the graph points
+   std::vector<Int_t> index(fNpoints);
```

Comment:
So in principle a vector of indices would be better described by a `std::vector<ULong64_t>`. But, the `Int_t` type is imposed by the signature of the comparison operator input argument `Bool_t (*greaterfunc)(const TGraph*, Int_t, Int_t)`. Can you please document this in the declaration of the vector?